bigframes/core/__init__.py

```diff
@@ -461,6 +470,16 @@ def shape(self) -> typing.Tuple[int, int]:
         width = len(self.columns)
         count_expr = self._to_ibis_expr(ordering_mode="unordered").count()
         sql = self._session.ibis_client.compile(count_expr)
+        api_methods_len = len(self._api_methods)
+
+        # Initialize methods to add as an empty list
+        add_api_methods = []
+        api_methods_len = len(self._api_methods)
+        if api_methods_len >= MAX_LABELS_COUNT:
```

Comment:
I don't think this is the correct layer to check checking for this limit. Rather, better to apply the limit at the time in which it matters (the labels on the job).

Also, the limit will be more appropriate to be applied when we create the job because it's possible we have other labels on the job besides these that we need to keep attached.

Reply:
Move the logic to `Session._start_query`